db/dcrpg/pgblockchain_charts_test.go

```diff
@@ -72,26 +39,45 @@ func TestMain(m *testing.M) {
 // It also checks the order and duplicates in the x-axis dataset.
 func TestPgCharts(t *testing.T) {
 	charts := cache.NewChartData(context.Background(), 0, &chaincfg.MainNetParams)
-	charts.DiffInterval = 10
+
 	db.RegisterCharts(charts)
+	// Register a dummy updater for Fees, PoolSize, and PoolValue. This must be
+	// registered *after* ChainDB registers its updaters, which set the correct
+	// length of blocks.Time and the other slices.
+	registerDummyFeeAndPoolInfo(charts)
+
+	// Validator for the Blocks and Windows chart data slice lengths.
 	blocks := charts.Blocks
 	windows := charts.Windows
 
 	validate := func(tag string) {
 		// Not checking NewAtoms right now, as the test database does not appear to
 		// contain stakebase and coinbase vins.
-		_, err := cache.ValidateLengths(blocks.Time, blocks.Chainwork, blocks.TxCount, blocks.BlockSize)
+		_, err := cache.ValidateLengths(blocks.Time, blocks.Chainwork,
+			blocks.TxCount, blocks.BlockSize, blocks.Fees, blocks.PoolSize,
+			blocks.PoolValue)
 		if err != nil {
 			t.Fatalf("%s blocks length validation error: %v", tag, err)
 		}
-		_, err = cache.ValidateLengths(windows.TicketPrice, windows.PowDiff, windows.Time)
+		_, err = cache.ValidateLengths(windows.TicketPrice, windows.PowDiff,
+			windows.Time, windows.StakeCount, windows.MissedVotes)
 		if err != nil {
 			t.Fatalf("%s windows length validation error: %v", tag, err)
 		}
 	}
 
+	t.Log("Validating initial state.")
+	validate("pre-update")
+
+	// Perform the DB queries and append charts data.
+	t.Log("Performing initial DB query and dataset updates.")
 	charts.Update()
+
+	// Ensure all the slice lengths generated by ChainDB's updaters are consistent.
+	t.Log("Validating initial update.")
 	validate("initial update")
```

Comment:
Since `(ChartData).Update` calls `Lengthen`, won't these always pass validation now? You could potentially return an `error` from `(ChartData).Update` to catch validations errors.

Reply:
That's true, although validating after `Snip` seems like a worth while check too.